protos/icmp/icmp.go

```diff
@@ -0,0 +1,329 @@
+package icmp
+
+import (
+	"net"
+	"time"
+
+	"github.com/elastic/libbeat/common"
+	"github.com/elastic/libbeat/logp"
+	"github.com/elastic/libbeat/publisher"
+
+	"github.com/elastic/packetbeat/config"
+	"github.com/elastic/packetbeat/protos"
+
+	"github.com/tsg/gopacket/layers"
+	// TODO: better use this? (also see commit #16691f7)
+	// "github.com/google/gopacket/layers"
+)
+
+type Processor interface {
+	ProcessICMPv4(tcphdr *layers.ICMPv4, pkt *protos.Packet)
+	ProcessICMPv6(tcphdr *layers.ICMPv6, pkt *protos.Packet)
+}
+
+// Contains all used information from the ICMP message on the wire.
+type IcmpMessage struct {
```

Comment:
for  example see memcache protocol. As long as I don't have to fulfil an interface I would prefer lower case symbols. If I want to export some symbol later gorename is my friend.

About naming (once private): do as it suits you best. Types "transaction" and "message" feel separate enough from other protocols. Still I would prefer "icmpTuple" over "tuple" as we've got other tuple types in libbeat/common (e.g. IPPortTuple).